app/src/main/java/com/example/studentarena/EndlessRecyclerViewScrollListener.java

```diff
@@ -0,0 +1,88 @@
+package com.example.studentarena;
+
+import androidx.recyclerview.widget.GridLayoutManager;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.recyclerview.widget.StaggeredGridLayoutManager;
+
+public abstract class EndlessRecyclerViewScrollListener extends RecyclerView.OnScrollListener{
+    // The minimum amount of items to have below your current scroll position
+    // before loading more.
+    private int visibleThreshold = 5;
+    // The current offset index of data you have loaded
+    private int currentPage = 0;
+    // The total number of items in the dataset after the last load
+    private int previousTotalItemCount = 0;
+    // True if we are still waiting for the last set of data to load.
+    private boolean loading = true;
+    // Sets the starting page index
+    private int startingPageIndex = 0;
+
+    RecyclerView.LayoutManager mLayoutManager;
+
+    public EndlessRecyclerViewScrollListener(LinearLayoutManager layoutManager) {
+        this.mLayoutManager = layoutManager;
+    }
+
+    public EndlessRecyclerViewScrollListener(GridLayoutManager layoutManager) {
+        this.mLayoutManager = layoutManager;
+        visibleThreshold = visibleThreshold * layoutManager.getSpanCount();
+    }
+
+    public EndlessRecyclerViewScrollListener(StaggeredGridLayoutManager layoutManager) {
+        this.mLayoutManager = layoutManager;
+        visibleThreshold = visibleThreshold * layoutManager.getSpanCount();
+    }
+
+    public int getLastVisibleItem(int[] lastVisibleItemPositions) {
+        int maxSize = 0;
+        for (int i = 0; i < lastVisibleItemPositions.length; i++) {
+            if (i == 0) {
+                maxSize = lastVisibleItemPositions[i];
+            }
+            else if (lastVisibleItemPositions[i] > maxSize) {
+                maxSize = lastVisibleItemPositions[i];
+            }
+        }
+        return maxSize;
```

Comment:
if i'm reading this right, you're looking for the biggest value in this `int` array, right? you should be able to do that in two lines:
- one line to convert your `int[]` into a `Integer[]` (this is a silly Java thing)
  - see [this StackOverflow question](https://stackoverflow.com/questions/880581/how-can-i-convert-int-to-integer-in-java) for how
- one to call `Collections.max()` on your new `Integer[]`
  - see [this site](https://www.geeksforgeeks.org/find-max-min-value-array-primitives-using-java/) for how

the difference between `int` and `Integer` in Java is kind of... weird to explain. basically `int` is just a barebones primitive number without any member functions or anything, and `Integer` is a fancy class that wraps that primitive `int` and gives it useful member functions like `toString()`. it also implements the functions that `Collections.max()` uses to check whether one value is larger than another value, which is why we need an `Integer[]` and not just an `int[]`. this fancy `Integer` class that wraps a primitive type is an example of a _boxed type_ - another example is `float`/`Float`. 

see if you can get this down to two lines using the links above, but if you can't get it working in like 15 minutes we can come back to it another time.